### PR TITLE
airbyte-ci: allow empty pull request number for `migrate-base-image` cmd

### PR DESCRIPTION
**Pull request by natikgadzhi**
Sometimes you don't yet have the PR, so supplying the PR number doesn't make sense in certain cases. This PR would allow omitting changelog entry, folks would have to do that manually. Happy to add test cases!

### CODE REVIEW

**Review by alafanechere — APPROVED** [review]
I'm in for this change - added minor suggestions. I'm wondering if we should use a placeholder for PR number instead of not writing the entry at all.
The missing changelog entry will get caught by CI so I don't really mind the final behavior you'll pick.
Would you consider making the same change to the `bump_version` command which has the same annoying requirement for a PR number?

And please don't forget to bump the version in `pyproject.toml` and add a changelog entry in README :) (sorry, no autobump for this package)

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_base_image/commands.py [hunk 1]
```diff
@@ -14,13 +15,16 @@
...
 @click.pass_context
 async def migrate_to_base_image(
     ctx: click.Context,
-    pull_request_number: str,
+    pull_request_number: str | None,
 ) -> bool:
-    """Bump a connector version: update metadata.yaml, changelog and delete legacy files."""
+    """
+    Bump a connector version: update metadata.yaml, changelog and delete legacy files.
+    If the `PULL_REQUEST_NUMBER` is not provided, no changelog entry will be added.
```
[the remark this answers is not in the document]
Comment (natikgadzhi) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_base_image/commands.py:26]: The problem with that is downstream code does `int(pull_request_number)` — I'm afraid to pull too much on that thread. I'm just learning how that code works!

**Comment by natikgadzhi** [comment]
oh.. Running local formatting requires docker! 🙈

**Comment by natikgadzhi** [comment]
@alafanechere please teach me to run formatters locally and let's make sure it's in the root-level CONTRIBUTING or something. I just ran `poetry run isort` something something at the root of the repo and it edited a bunch of files. Weird. 

Anyway, done with feedback here.

**Comment by alafanechere** [comment]
> @alafanechere please teach me to run formatters locally and let's make sure it's in the root-level CONTRIBUTING or something. I just ran `poetry run isort` something something at the root of the repo and it edited a bunch of files. Weird.
> 
> Anyway, done with feedback here.

It is documented here https://docs.airbyte.com/contributing-to-airbyte/resources/code-formatting

**Comment by alafanechere** [comment]
@natikgadzhi thanks for the changes. You might also want to update the [documentation](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md#L520) of the command you changed

**Comment by natikgadzhi** [comment]
Fixed version, fixed readme, and fixed a lint error.

**Pull request merged**